protocol-impl/src/main/java/io/zeebe/protocol/impl/record/RecordMetadata.java

```diff
@@ -202,6 +236,7 @@ public RecordMetadata reset() {
     intent = null;
     rejectionType = RejectionType.NULL_VAL;
     rejectionReason.wrap(0, 0);
+    brokerVersion = CURRENT_BROKER_VERSION;
```

Comment:
By default, it set the current version. So, we don't need to set the version explicitly when using the log stream writer.